producer/src/main/java/com/flipkart/varadhi/produce/services/ProducerService.java

```diff
@@ -177,17 +210,19 @@ private CompletableFuture<ProduceResult> produceToValidTopic(
      * This method first checks if the producer is already in the cache. If not, it attempts
      * to load it using the producer provider function.
      *
-     * @param storageTopic the storage topic to get a producer for
+     * @param storageTopic     the storage topic to get a producer for
+     * @param varadhiTopicName the name of the Varadhi topic (used for caching)
      * @return a future that completes with the producer
      */
-    public CompletableFuture<Producer> getProducer(StorageTopic storageTopic) {
-        Producer producer = producerCache.getIfPresent(storageTopic);
+    public CompletableFuture<Producer> getProducer(StorageTopic storageTopic, String varadhiTopicName) {
```

Comment:
fix the order of param here as well. 
varadhiTopicName first and then the storage topic name within it.